src/fraction.cpp

```diff
@@ -122,7 +122,7 @@ void Fraction::Reduce()
         m_numerator = -m_numerator;
         m_denominator = -m_denominator;
     }
-    int gcdVal = std::gcd(abs(m_numerator), abs(m_denominator));
+    const int gcdVal = std::gcd(m_numerator, m_denominator);
```

Comment:
This is OK, because `std::gcd()` (introduced in C++17) internally applies `abs()` to the two input arguments (or at least only outputs non-negative values).   There is also a GCC macro `__gcd()` that can return negative values:

```cpp
#include <iostream>
#include <algorithm>     // For std::gcd (C++17 and later)
#include <bits/stdc++.h> // For __gcd    (GCC specific)

int main() {
    int a = -2, b = 4;
    std::cout << std::gcd(a, b) << std::endl;  // +2
    std::cout << __gcd(a, b) << std::endl;     // -2
    return 0;
}
```